packages/block-library/src/post-comments-form/view.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import './runtime/init.js';
+import { store, navigate } from './runtime';
+
+store( {
+	state: {
+		core: {
+			commentsFormError: '',
+		},
+	},
+	actions: {
+		core: {
+			commentsFormSubmission: async ( { event, state } ) => {
+				event.preventDefault();
+
+				state.core.commentsFormError = '';
+
+				const formData = new FormData( event.target );
+
+				const res = await fetch(
+					'http://localhost:8888/wp-comments-post.php',
+					{
+						method: 'POST',
+						body: formData,
+					}
+				);
+
+				const html = await res.text();
+
+				if ( res.status !== 200 ) {
+					const dom = new window.DOMParser().parseFromString(
+						html,
+						'text/html'
+					);
+					state.core.commentsFormError =
+						dom.querySelector( 'p' ).innerHTML;
```

Comment:
This might be a bit more robust:

```suggestion
						dom.querySelector( '.wp-die-message' ).innerHTML;
```

Not only would it preserve the paragraph in the response (instead of rendering the text in a bare `div`), but it would also account for there possibly being additional paragraphs or other elements rendered on the comment error screen.